plugins/inputs/tacacs/tacacs_test.go

```diff
@@ -0,0 +1,353 @@
+			usedSecret:           config.NewSecret([]byte(`WRONGSECRET`)),
+			requestAddr:          "127.0.0.1",
+			expectNoGatherErrors: false,
+		},
+		{
+			name:                 "unreachable",
+			testingTimeout:       config.Duration(time.Nanosecond * 1000),
+			serverToTest:         []string{"unreachable.hostname.com:404"},
+			usedUsername:         config.NewSecret([]byte(`testusername`)),
+			usedPassword:         config.NewSecret([]byte(`testpassword`)),
+			usedSecret:           config.NewSecret([]byte(`testsecret`)),
+			requestAddr:          "127.0.0.1",
+			expectNoGatherErrors: false,
+		},
+	}
+
+	for _, tt := range testset {
+		t.Run(tt.name, func(t *testing.T) {
+			plugin := &Tacacs{
+				ResponseTimeout: tt.testingTimeout,
+				Servers:         tt.serverToTest,
+				Username:        tt.usedUsername,
+				Password:        tt.usedPassword,
+				Secret:          tt.usedSecret,
+				RequestAddr:     tt.requestAddr,
+				Log:             testutil.Logger{},
+			}
+
+			var acc testutil.Accumulator
+
+			require.NoError(t, plugin.Init())
+			require.NoError(t, plugin.Gather(&acc))
+
+			if tt.expectNoGatherErrors {
+				require.Len(t, acc.Errors, 0)
+				if !acc.HasMeasurement("tacacs") {
+					t.Errorf("acc.HasMeasurement: expected tacacs")
+				}
+				require.Equal(t, true, acc.HasTag("tacacs", "source"))
+				require.Equal(t, srvLocal, acc.TagValue("tacacs", "source"))
+				require.Equal(t, true, acc.HasInt64Field("tacacs", "responsetime_ms"))
+				require.Equal(t, true, acc.HasTag("tacacs", "response_code"))
+			} else {
+				require.Len(t, acc.Errors, 1)
+				require.Equal(t, false, acc.HasTag("tacacs", "source"))
+				require.Equal(t, false, acc.HasInt64Field("tacacs", "responsetime_ms"))
+				require.Equal(t, false, acc.HasTag("tacacs", "response_code"))
+			}
+			if tt.name == "success_timeout_0s" {
+				require.Equal(t, strconv.FormatUint(uint64(tacplus.AuthenStatusPass), 10), acc.TagValue("tacacs", "response_code"))
+			}
+			if tt.name == "wrongpw" {
+				require.Equal(t, strconv.FormatUint(uint64(tacplus.AuthenStatusFail), 10), acc.TagValue("tacacs", "response_code"))
+				require.Equal(t, time.Duration(plugin.ResponseTimeout).Milliseconds(), acc.Metrics[0].Fields["responsetime_ms"])
+			}
+			if tt.name == "wrongsecret" {
+				require.ErrorContains(t, acc.Errors[0], "error on new tacacs authentication start request to "+srvLocal+" : bad secret or packet")
+			}
+			if tt.name == "unreachable" {
+				require.ErrorContains(t, acc.Errors[0], "error on new tacacs authentication start request to unreachable.hostname.com:404 : dial tcp")
+			}
+		})
+	}
+}
+
+func TestTacacsInit(t *testing.T) {
+	var testset = []struct {
+		name               string
+		testingTimeout     config.Duration
+		serverToTest       []string
+		usedUsername       config.Secret
+		usedPassword       config.Secret
+		usedSecret         config.Secret
+		requestAddr        string
+		expectNoInitErrors bool
```

Comment:
What would you think of this?
```suggestion
		plugin      Tacacs
		errContains string
```
(+ next suggestion)